lib/parsers/gjs-gts-parser.js

```diff
@@ -533,12 +538,88 @@ module.exports = {
     let jsCode = code;
     const info = transformForLint(code);
     jsCode = info.output;
+    jsCode = jsCode.replaceAll(/\.gts(["'])/g, '.ts$1 ');
 
     const isTypescript = options.filePath.endsWith('.gts');
 
     let result = null;
+
+    const projectFile = resolve(options.tsconfigRootDir, options.project);
+
+    const config = ts.getParsedCommandLineOfConfigFile(
+      projectFile,
+      {},
+      {
+        ...ts.sys,
+        readDirectory(dir) {
+          const results = ts.sys.readDirectory(dir);
+          return results.map((f) => f.replace(/\.gts$/, '.ts'));
+        },
+        onUnRecoverableConfigFileDiagnostic(diagnostic) {
+          let { messageText } = diagnostic;
+          if (typeof messageText !== 'string') {
+            messageText = messageText.messageText;
+          }
+
+          throw new Error(messageText);
+        },
+      }
+    );
+
+    const host = ts.createCompilerHost(config.options);
+    host.fileExists = function (fileName) {
+      try {
+        return fs.existsSync(fileName.replace(/\.ts$/, '.gts'));
+      } catch {
+        return fs.existsSync(fileName);
+      }
+    };
+
+    const readDirectory = host.readDirectory;
+    host.readDirectory = function(dir) {
+      const results = readDirectory.call(this, dir);
+      return results.map((f) => f.replace(/\.gts$/, '.ts'));
+    };
+    host.readFile = function(fileName) {
+      if (fileName === options.filePath.replace(/\\/g, '/').replace(/\.gts$/, '.ts')) {
+        return jsCode;
+      }
+      let content = fs.readFileSync(fileName).toString();
```

Comment:
I think `readFileSync` won't be able to find the file on disk as it will have `ts` extensino, we need to map this back to `.gts` in the fileName

Reply:
Yeah, it's encroaching on re-inventing @glint/core

Reply:
Glint/core does maaaybe need to export the program tho.

See here for some background 
https://github.com/typed-ember/glint/issues/608

Reply:
Yes, I'm looking at https://github.com/typed-ember/glint/blob/main/packages/core/src/cli/perform-check.ts
If this doesn't work, I might give it a try to just copy that then.
but there is also the performance part. Glint would do a lot more than what we need.

Reply:
I we can just paste glint and have it work, that would be great.
But I do have a concern regarding ESTREE -> TS mapping
See https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-estree/src/parser.ts#L248-L261
typescript-eslint sort of takes the AST generated from typescript and maps it to an ESTREE.
When rules are evaluated we use NODES from the ESTREE, pass them to the typechecker service generated by typescript-eslint which then useses the Map to get back the TS node based on ESTREE key we provided.
But I'm not sure if we will get missmatches because glint reads .gts files differently then gjs-gts parse, or to be more precise I assume the outputs don't match.
This is just a theory by the way after doing some analysis on the code

Reply:
btw @patricklx I managed to get type awareness working by fixing what I commented initially in this thread, when importing from `.gts` typechecker was working fine, but I somehow broke imports from regular .ts files,  and type aware imports from `.ts` files stoped working 🤯🤯🤯

Reply:
Ohh! Interesting. I will look at it tomorrow

Reply:
@patricklx @NullVoxPopuli also came across this repo https://github.com/ota-meshi/typescript-eslint-parser-for-extra-files/tree/main it seems to try to do the same thing for vue and svelte, maybe it will be usefull for somone to get some ideas or something